flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/hive/HiveDDLUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.parser.ddl.hive;
+
+import org.apache.flink.sql.parser.ddl.SqlTableOption;
+import org.apache.flink.sql.parser.impl.ParseException;
+import org.apache.flink.table.catalog.config.CatalogConfig;
+
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.apache.flink.sql.parser.ddl.hive.SqlAlterHiveDatabase.ALTER_DATABASE_OP;
+import static org.apache.flink.sql.parser.ddl.hive.SqlCreateHiveDatabase.DATABASE_LOCATION_URI;
+
+/**
+ * Util methods for Hive DDL Sql nodes.
+ */
+public class HiveDDLUtils {
+
+	private static final Set<String> RESERVED_DB_PROPERTIES = new HashSet<>();
+
```

Comment:
It seems hacky we put these properties internal through the table options, i think these properties should be kept in each SqlNode but not the property list.

Reply:
I agree it seems hacky. But the problem is some semantics are specific to Hive and may never be adopted in Flink, e.g. the locations of table and database. If we keep them in the SqlNode, the planner needs to be able handle them, and the Catalog object (e.g. CatalogDatabase) needs extra fields for the semantics. To avoid this, we have to encode the Hive-specific semantics as properties, which has been mentioned and discussed in FLIP-123.

Reply:
As discussed offline, this is by-design intentionally, previously we design the Flink databases DDLs to have a extensible properties for such "hacky" extention way to make the planner code clean.

While for the long term, if we also want to make a Hive dialect DML, we may need another planner which is another big story.